### DISCUSSION TITLE
require 2 fingers to scroll?

### DISCUSSION THREAD

**Discussion by ImageDeeply** [rxhanson/Multitouch-Community#201] [Q&A]
I recently switched to the Magic Mouse. Being able to scroll is sometimes useful but:
1. I want it to require 2 fingers not just 1
2. ideally make horizontal scroll even harder, e.g. 3 fingers or less sensitive or some such

The Multitouch app UI seems to only have 'hold and click' actions. Did I miss something? Or is there another app that handles item 1 (and maybe 2)?

Thanks.

**Comment by rxhanson** [comment]
Thanks for asking. At the moment, I don't have an app with specifically this functionality. The closest thing I have is [Scroll](https://ryanhanson.dev/scroll), where you could limit the area of the Magic Mouse where scrolling is allowed.

**Comment by ImageDeeply** [comment]
OK. I downloaded Scroll. I have new Apple Silicon MacBook running latest Sonoma 14.4. Are there some common apps where you know it does or doesn't work? e.g. Safari, Chrome, Firefox ... BBEdit ... Finder, TextEdit, Preview

Related: do you have a User Guide? I assume it's supposed to be self-explanatory, but the behavior isn't clear to me. Trial & error might suffice ... if I knew which apps to test in!

I don't need horizontal scroll, so will always check 'disable horizontal scroll'

If I set 'Allow' 'scroll while holding' the ctrl key ... does that mean it should ONLY scroll when I hold the ctrl key? If so, then maybe the app does nothing on my machine for any of the above apps? (I completely understand that's likely out of your control. Apple often makes it impossible for devs to add 'missing' features.)

I don't understand 'allow scroll in area'.
- does that only apply to the remaining vertical scroll?
- if the slider is in the middle, is the 'allowed' area the top or the bottom?
- why a slider? Seems like some people would want top, others bottom.

**Reply by rxhanson** [reply]
> OK. I downloaded Scroll. I have new Apple Silicon MacBook running latest Sonoma 14.4. Are there some common apps where you know it does or doesn't work? e.g. Safari, Chrome, Firefox ... BBEdit ... Finder, TextEdit, Preview

It has been reported to not work in some of the Microsoft apps, like Excel, although for some users it appears to work fine. I can reproduce it in Safari. 

> Related: do you have a User Guide? I assume it's supposed to be self-explanatory, but the behavior isn't clear to me. Trial & error might suffice ... if I knew which apps to test in!

No user guide, sorry. Indeed the idea is that is self-explanatory, and hopefully we can clear up confusion here.

> If I set 'Allow' 'scroll while holding' the ctrl key ... does that mean it should ONLY scroll when I hold the ctrl key? 

Yes

> If so, then maybe the app does nothing on my machine for any of the above apps? (I completely understand that's likely out of your control. Apple often makes it impossible for devs to add 'missing' features.)

If you don't hold the control key, then the app will attempt to remove scrolling.

> I don't understand 'allow scroll in area'.
> does that only apply to the remaining vertical scroll?
> if the slider is in the middle, is the 'allowed' area the top or the bottom?

For whichever dimension(s) you've checked boxes for, the app will attempt to remove scrolling when your touch is above the slider handle. Conversely, it will let the scrolling pass through when your touch is below the slider handle (the highlighted area of the slider).

> why a slider? Seems like some people would want top, others bottom.

For simplicity. If you really want to change the "allow" area to something that's not in the UI, then you can do so with a terminal command:

```
defaults write com.knollsoft.Scroll allowRectangle -string "[[0,0.5],[1,0.5]]"
```
where the values you're adding are [[x,y],[w,h]]

You'll need to restart the app after executing that command.

If you have this configured, then the slider for allowing scroll area in the UI is not taken into account when the app determines whether or not to reject scroll.
